src/test/java/com/fittracker/fittracker/controller/WeightControllerIntegrationTest.java

```diff
@@ -95,25 +98,78 @@ void givenValidPostRequestWithExistingDate_shouldReturnError() throws Exception
     @Nested
     class Get {
 
-        @Test
-        void givenValidGetRequestWithExistingDate_shouldReturnWeight() throws Exception {
-            weightRepository.save(new Weight(LocalDate.of(2023, 8, 3), 13.2, testUUID));
-            assertThat(weightRepository.findAll()).hasSize(1);
+        @Nested
+        class WeightEndpoint {
+            @Test
+            void givenValidGetRequestWithExistingDate_shouldReturnWeight() throws Exception {
+                weightRepository.save(new Weight(LocalDate.of(2023, 8, 3), 13.2, testUUID));
 
-            mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
-                            .header("Authorization", "Bearer " + TOKEN))
-                    .andExpect(status().isOk())
-                    .andExpect(content().string("{\"date\":\"2023-08-03\",\"value\":13.2}"));
+                mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
+                                .header("Authorization", "Bearer " + TOKEN))
+                        .andExpect(status().isOk())
+                        .andExpect(content().string("{\"date\":\"2023-08-03\",\"value\":13.2}"));
+
+                assertThat(weightRepository.findAll()).hasSize(1);
+            }
+
+            @Test
+            void givenValidGetRequestWithNonexistentDate_shouldReturnError() throws Exception {
+
+                mockMvc.perform(get(URI.create(ENDPOINT + "?date=2023-08-03"))
+                                .header("Authorization", "Bearer " + TOKEN))
+                        .andExpect(status().isNotFound())
+                        .andExpect(content().string("{\"message\":\"Weight not found for date: 2023-08-03\"}"));
+            }
         }
 
-        @Test
-        void givenValidGetRequestWithNonexistentDate_shouldReturnError() throws Exception {
+        @Nested
+        class WeightsEndpoint {
+            private final ObjectMapper mapper = new ObjectMapper();
+
+            @BeforeEach
+            public void beforeEach() {
+                List<Weight> weights = List.of(
+                        new Weight(LocalDate.of(2023, 1, 1), 100.0, testUUID),
+                        new Weight(LocalDate.of(2023, 1, 10), 95.5, testUUID),
+                        new Weight(LocalDate.of(2023, 2, 20), 98.2, testUUID)
+                );
+                weights.forEach(weight -> weightRepository.save(weight));
+            }
+
+            @Test
+            void givenDateRange_shouldReturnListOfWeightResponses() throws Exception {
+                mockMvc.perform(get(URI.create(ENDPOINT + "s?startDate=2023-01-01&endDate=2023-03-04"))
```

Comment:
How about we change it to `startDate=2023-01-02`, so we get 2 elements in the response?

Reply:
Good idea. This way we are ensuring that it's not just all elements.